packages/autorest.python/autorest/codegen/models/enum_type.py

```diff
@@ -168,14 +169,22 @@ def description(
         enum_description = f"Known values are: {possible_values_str}."
         return enum_description
 
+    def typing_name(self, need_module_name: bool = True) -> str:
+        return typing_name(
+            file_name=self.code_model.enums_filename,
+            internal=self.internal,
+            need_module_name=need_module_name,
+            type_name=self.name,
+        )
+
     def type_annotation(self, **kwargs: Any) -> str:
         """The python type used for type annotation
 
         :return: The type annotation for this schema
         :rtype: str
         """
         if self.code_model.options["models_mode"]:
-            model_name = f"_models.{self.name}"
+            model_name = self.typing_name()
```

Comment:
seems you want to abstract the type typing name, but not cover all the branches.

Reply:
For now, only enum type and model type need fix.